pkg/scheduler/core/generic_scheduler.go

```diff
@@ -673,12 +663,15 @@ func podFitsOnNode(
 			if predicate, exist := predicateFuncs[predicateKey]; exist {
 				fit, reasons, err = predicate(pod, metaToUse, nodeInfoToUse)
 				if err != nil {
-					return false, []predicates.PredicateFailureReason{}, err
+					return false, []*framework.Status{}, err
 				}
 
 				if !fit {
 					// eCache is available and valid, and predicates result is unfit, record the fail reasons
-					failedPredicates = append(failedPredicates, reasons...)
+					for _, reason := range reasons {
+						// FIXME
+						failedPredicates = append(failedPredicates, framework.NewStatus(framework.Error, reason.GetReason()))
```

Comment:
If `!fit`, it means that pod is unschedulable. So, the status code should be `Unschedulable`.

Reply:
Done